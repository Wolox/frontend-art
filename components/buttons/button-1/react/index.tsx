import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
