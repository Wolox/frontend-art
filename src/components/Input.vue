<template lang="pug">
  .input-section-container
    #input-1.input-container
      span.input-subtitle Input 1
      .input-wrapper
        input.input-base(type='email')
        label.input-label Email
    #input-2.input-container
      span.input-subtitle Input 2
      label.inp(for='inp')
        input#inp(type='text' placeholder='Â ')
        span.label Label
        span.border
    #input-3.input-container
      span.input-subtitle Input 3
      .Input
        input#input.Input-text(type='text' placeholder='Your first name, e.g. Nicholas')
        label.Input-label(for='input') First name
    #input-4.input-container
      span.input-subtitle Input 4
      .Input
        .form-wrap
          .input-container
            label.label4(for='mail') Email address
            input.input4(type='email' name='mail' autocomplete='off')
          svg.line.svg4
</template>
<script>
import '../../scripts/input'
export default {

}
</script>

<style lang="scss" scoped>
@import './input/scss/input1';
@import './input/scss/input2';
@import './input/scss/input3';
@import './input/scss/input4';

.input-section-container {
  padding: 40px;
}

.input-container {
  display: flex;
  flex-direction: column;
  margin-bottom: 40px;
}

.input-subtitle {
  font-size: 14px;
  text-transform: uppercase;
  margin-bottom: 20px;
}
</style>

