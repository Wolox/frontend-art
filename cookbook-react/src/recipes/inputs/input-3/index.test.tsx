import React from 'react';
import { render } from '@testing-library/react';

import Input from './index';

describe('Input-3', () => {
  test('match snapshot', () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  test('Input is disabled', async () => {
    const props = { label: 'Label', disabled: true, error: '' };
    const { findByRole } = render(<Input {...props} />);

    const inputElement = await findByRole('textbox');
    const disabledAttribute = inputElement?.getAttribute('disabled');

    expect(disabledAttribute).not.toBe(null);
    expect(disabledAttribute).not.toBe(false);
  });

  test('shows the error message', () => {
    const props = { label: 'Label', disabled: true, error: 'Some error' };
    const { getByRole } = render(<Input {...props} />);

    const errorElement = getByRole('alert');

    expect(errorElement).toBeInTheDocument();
  });
});
