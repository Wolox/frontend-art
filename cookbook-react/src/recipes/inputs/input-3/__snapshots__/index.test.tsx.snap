// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input-3 match snapshot 1`] = `
<div>
  <div
    id="input-3"
  >
    <div
      class="input-wrapper"
    >
      <input
        class="input-base"
        id="input-tag-3"
        placeholder=" "
      />
      <label
        class="input-label"
        for="input-tag-3"
      >
        Label
      </label>
    </div>
  </div>
</div>
`;
